components/fxa-client/src/device.rs

```diff
@@ -149,26 +149,31 @@ impl FirefoxAccount {
             return Ok(Vec::new());
         }
         log::info!("Handling {} messages", pending_commands.messages.len());
-        let account_events = self.parse_commands_messages(pending_commands.messages)?;
+        let device_commands = self.parse_commands_messages(pending_commands.messages)?;
         self.state.last_handled_command = Some(pending_commands.index);
-        Ok(account_events)
+        Ok(device_commands)
     }
 
-    fn parse_commands_messages(&self, messages: Vec<PendingCommand>) -> Result<Vec<AccountEvent>> {
-        let mut account_events: Vec<AccountEvent> = Vec::with_capacity(messages.len());
+    fn parse_commands_messages(
+        &self,
+        messages: Vec<PendingCommand>,
+    ) -> Result<Vec<IncomingDeviceCommand>> {
+        let mut device_commands: Vec<IncomingDeviceCommand> = Vec::with_capacity(messages.len());
         let commands: Vec<_> = messages.into_iter().map(|m| m.data).collect();
         let devices = self.get_devices()?;
         for data in commands {
             match self.parse_command(data, &devices) {
-                Ok((sender, tab)) => account_events.push(AccountEvent::TabReceived((sender, tab))),
+                Ok((sender, payload)) => {
+                    device_commands.push(IncomingDeviceCommand::SendTab { sender, payload })
+                }
                 Err(e) => log::error!("Error while processing command: {}", e),
             };
         }
-        Ok(account_events)
+        Ok(device_commands)
     }
 
     // Returns SendTabPayload for now because we only receive send-tab commands and
-    // it's way easier, but should probably return AccountEvent or similar in the future.
+    // it's way easier, but should probably return IncomingDeviceCommand in the future.
```

Comment:
I agree with the sentiment behind this comment :-)

I wonder if there should be a corresponding comment on the code above that assumes the result is always a send-tab payload, since that's where I was when I notice that this behaviour was happening.

Alternately, I wonder if this is as good an opportunity as we'll get to go ahead and refactor it to return an `IncomingDeviceCommand`. I think it really would improve the legibility of this section of the code.